app/lib/service/download_counts/download_counts.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:basics/basics.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pub_semver/pub_semver.dart';
+part 'download_counts.g.dart';
+
+@JsonSerializable(includeIfNull: false)
+class CountData {
+  // We store at most two years of data, one data point per day.
+  static const countsCapacity = 731;
+  final versionCounts = <({String versionRange, List<int> counts})>[];
+  final patchCounts = <({String versionRange, List<int> counts})>[];
+  final minorCounts = <({String versionRange, List<int> counts})>[];
+  final majorCounts = <({String versionRange, List<int> counts})>[];
+  final totalCounts = <int>[];
```

Comment:
Should this not default to `List.filled(countsCapacity, 0)` or something like that?

Reply:
Also didn't we want to add a boolean list for tracking which days we do have data from? instead of using -1.

Reply:
Yes, the plan is to keep a bit vector as well. Not included in this PR.